### DISCUSSION TITLE
How to setup 1-n and 1-1 relationship simultaneously

### DISCUSSION THREAD

**Discussion by SharadKumar** [prisma/prisma#3850] [Q&A] [answered]
Lets say:
- Product has many Bids (`bids`)
- Product has one Winning Bid (`wonBy`)
- Bid belongs to one Product

```prisma
model Product {
  id          String        @default(cuid()) @id
  number      String?
  title       String?
  estimate    Float?
  reserve     Float?
  realised    Float?
  wonById     String        @map("bid_id")
  wonBy       Bid           @relation(name: "wonBy", fields: [wonById], references: [id])
  bids        Bid[]
}

model Bid {
  id         String    @default(cuid()) @id
  currency   String
  amount     Float
  status     BidStatus @default(SUBMITTED)
  createdAt  DateTime  @default(now()) @map("created_at")
  customerId String?   @map("customer_id")
  productId  String    @map("product_id")
  account    Account   @relation(fields: [accountId], references: [id])
  customer   Customer? @relation(fields: [customerId], references: [id])
  product    Product       @relation(fields: [productId], references: [id])
  products   Product[]     @relation("wonBy") // <<<<<< THIS IS NOT CORRECT <<<<<<<<<<
}
```

Whats the best way to do both relationships together? `Bid.products` is not correct since one Bid can't have multiple Products and need 1-1... ideally I need just `Bid.product` already there.

**Comment by ryands17** [comment, marked as answer]
@SharadKumar 👋 

Here you go:

```prisma
model Product {
  id         String  @id @default(cuid())
  number     String?
  title      String?
  estimate   Float?
  reserve    Float?
  realised   Float?
  bids       Bid[]   @relation("bidsList")
  winningBid Bid     @relation("winningBid")
}

model Bid {
  id              String   @id @default(cuid())
  currency        String
  amount          Float
  createdAt       DateTime @default(now()) @map("created_at")
  product         Product? @relation("bidsList", fields: [productId], references: [id])
  productId       String?
  winnerOfProduct Product? @relation("winningBid", fields: [winnerProductId], references: [id])
  winnerProductId String?
}
```

This should work as per your definition. Best to assign a relation name whenever there's more than 1 relation to the same model.

**Reply by SharadKumar** [reply]
Thank you @ryands17 😊

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
